### ISSUE TITLE
Proposal: Update design of TreeView

### ISSUE THREAD

**Issue by anawishnoff** [microsoft/microsoft-ui-xaml#4305]
# Proposal: Update design of TreeView 

## Summary

I'm proposing design changes to TreeView so that it visually aligns with the [new ListView designs](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/anawish/ListViewGridViewDesignUpdates/active/ListViewGridView/listviewgridview-designupdates.md). This includes providing items with a rounded backplate in non-rest states, an accent-colored selection indicator, and a more rounded checkbox in multiple selection mode. 

## Rationale

* As ListView is getting a design update, controls that have list functionality should align with that update.
* Many controls in WinUI have rounded corners, so TreeView should follow that trend and embrace the modern, Fluent look.

## Scope

| Capability | Priority |
| :---------- | :------- |
| TreeView items will have a rounded backplate in non-rest states. | Must |
| TreeView items should use a pill-shaped selection indicator in single selection mode. | Must |
| TreeView items should use the same rounded checkbox visual in multiple selection mode as ListView.  | Must |
| TreeView items should align with new ListView design details overall. | Must |
| These changes will not affect the functionality of TreeView. | Won't |

### Design Mockups
Below: TreeView in single selection mode.

![image](https://user-images.githubusercontent.com/22504341/108868167-ef5f3a00-75c3-11eb-92d6-17e04a793cd2.png)

Below: TreeView in multiple selection mode. 

![image](https://user-images.githubusercontent.com/22504341/108868218-fb4afc00-75c3-11eb-853f-36f9c2f2ab04.png)

## Open Questions

- Any existing TreeView use cases that would be broken by this design change?

**Comment by mdtauk** [comment]
Many of the other controls have moved to using Fluent UI System Icons, will the chevrons be replaced with caret symbols, to fit in with the other controls?

![image](https://user-images.githubusercontent.com/7389110/108889481-569cde80-7604-11eb-9e66-f0dbb326b48b.png)

![image](https://user-images.githubusercontent.com/7389110/108889650-88ae4080-7604-11eb-873e-681f533e00fc.png)
![image](https://user-images.githubusercontent.com/7389110/108889672-8e0b8b00-7604-11eb-9c33-21338d594e0d.png)

**Comment by anawishnoff** [comment]
@mdtauk Hmm, this is an interesting question/idea. I think the carets, as shown in the examples above, are usually used for scrolling or navigating purposes? At least that's where I've seen them. Chevrons are used for expanding/opening/closing items (i.e. Hierarchical NavigationView), so I think they're appropriate for TreeView as well.

**Comment by jschwizer99** [comment]
@anawishnoff , I appreciate a lot the step taken to align the visual style between listview and treeview. On the current preview4 the differences are quite noticeable.

Our WinUI application will target the classical desktop market. The treeview and listview are very important design elements to reflect the complex data of the 3D application. Therefore, we strongly rely on a clear distinction between multi-selection and mouse hover. In the new proposal, the contrast between selected elements and background is very weak if the checkboxes are disabled (`IsMultiSelectCheckBoxEnabled="False"`).
For applications targeting mainly the non-touch markets, disabling the checkbox display for the listview can be beneficial as we get display space for icons. At the moment I couldn't find an easy way to achieve the same for the treeview. Thus, an alignment of the visual style should also ensure that the layout configuration possibilities are the same. If it is possible to adjust the layout and style to match the customer's needs it would be great.

Based on the above explanation:
- Allow disabling of checkbox display for multi-select for both treeview and listview (`IsMultiSelectCheckBoxEnabled="False"`)
- Support XAML based configuration of select and hover styles to match with target application and market

**Comment by anawishnoff** [comment]
This is great feedback @jschwizer99, thank you! 

This proposal won't be covering any actual functionality changes, as mentioned above, so allowing the disabling of checkbox for multiple selection mode in TreeView won't be included in this change. However, this is definitely an interesting topic. I'd recommend opening up a separate issue for this. I agree and don't see why you can disable checkboxes with ListView, but not TreeView.

If you're looking to change the colors for selection and hover styles, you'll still be able to do that by editing the control's template and providing the colors you'd like to show. However, I do appreciate the feedback about the contrast - I'm going to take a better look into this.

**Comment by Noemata** [comment]
Some design choices may appear heavy handed at first (e.g. prominent selection regions), but such choices do help those that have partial visual impairment.  UX design aesthetics have practical usability considerations that go beyond a given fashionable design paradigm.  Does WinUI have a UX design team with appropriate UX usability credentials that include disability awareness as well as current UX design know how?  Typically, a UX mockup for a specific element sits beside a bunch of other UI components so the overall look and feel can be evaluated.

**Comment by anawishnoff** [comment]
@Noemata Yes, we work with an awesome design team who has vast knowledge of usability as well as accessibility.

**Comment by Noemata** [comment]
@anawishnoff , that's cool.  You might want to surface their designs here for this sort of change.

**Comment by llothar** [comment]
I really not happy with the simplicity of this TreeView for Desktop apps.  
I have to agree with @jschwizer99  on this one.

But i already knew i have to roll out my own immediately as i start porting our app to WinUI3. I just hope that the TreeView   code will soon be an open source released and we can extend it as much as they we want/need.

**Comment by llothar** [comment]
The other issue i have is that you only use two arrow indicators. But we have three states for each tree item
a) collapsed
b) expanded
c) collapsed (because it's empty)

If you search for existing "outliners" your design team will find that there are a lot of well known solutions for this. 
Also imagine you want to show a file browser in the tree view with combined files and directories. Do you really want to show an ">" icon in front of each file? While technically each view item can have children, conceptionally that might be not the case.

The check box with the small low contrast "-" (for some children selected) and "v" for this item selected is also way to close in look and therefore confusing. Again, best to get fully rid of this ugly boxes.  But if you have then, can you give the "-" checkbox for example a lighter blue or some other clue to recognize them more easily.

**Comment by niels9001** [comment]
Closing this issue as the proposed designs have been implemented. Please open a new issue if you would like to discuss any changes to the new styles.
